### PR TITLE
[feat] adds un- states to the interest form

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
Adds un- states to the interest form
### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Utilizes `ProfileContext` to conditionally render the the interest form or blocked forms on the interest form depending on whether a user is logged in and has been onboarded
- Refactors styling on `InterestForm` (substituting `display: block` & fixing margins, paddings, etc)
- Utilizes `ProfileContext` to add the current logged-in user's user ID to the interest form submission

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

**Cases Present, unauthenticated:**
image

**Cases Present, un-onboarded:**
image

**Cases Present, onboarded:**
image

**No Cases Present (filtered):**
image

**No Cases Present (no cases in database):**
image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- fetch & checkout branch, `npm run dev`, and head to the `cases` page

- **onboarded account**
email: berrybumpy@gmail.com
password: test123456

- **un-onboarded account**
email: ccatherinetan100222@berkeley.edu
password: hi1234

**Cases Present, unauthenticated:**
- the cases page should look like the corresponding screenshot for this scenario since you are not logged in yet

**Cases Present, un-onboarded:**
- log in using the credentials from the un-onboarded account above (either press login on the cases page, or navigate to `login` using the url to log in)
- The cases page should look like the corresponding screenshot for this scenario after loggin in to the un-onboarded account

**Cases Present, onboarded:**
- If you're already logged in to another account, you'll need to sign out first. Navigate to the `settings` page using the url or press on the profile icon at the top right of the page, and press the `Sign Out` button once you're in the profile/settings page.
- Similar to the previous case, now log in using the credential of the onboarded account
- The cases page should look like the corresponding screenshot for this scenario after logging in to the onboarded account

**No Cases Present (filtered):**
- For this, just filter the cases so that no cases are displayed in the `CardColumn`
- the cases page should look like the corresponding screenshot for this scenario after finding a filter combination that displays no cases

**No Cases Present (no cases in database):**
- uh if you reload you can see it for like 2 seconds lol, need to fix this later?? now?? idk
- However, if you wanna see it on the screen longer, head into `cases/index.tsx` file change `filteredCases.length === 0` to `true` in the ternary statement and `caseInfo` to `false` in the other ternary statement below 

Like dis:
image

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Fix this issue on reload 
showcassssee
- Add a loading state

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/components/CaseDetails/styles.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import COLORS from '@/styles/colors';
 export const CaseDisplay = styled.aside`
   position: relative;
   width: 100%;
-  margin-bottom: 0.5rem;
+  /* margin-bottom: 0.5rem; */ //idk if I need this
```
Comment (jinkang-0) [inline src/components/CaseDetails/styles.tsx]: you probably dont...

File: src/app/cases/page.tsx [hunk 2]
```diff
@@ -203,20 +204,35 @@ export default function Page() {
...
+                  }}
+                />
+              ))}
+            </>
+          )}
         </CardColumn>
         <CaseDetailsContainer>
-          {caseInfo && <CaseDetails caseData={caseInfo} />}
+          {caseInfo ? (
+            <CaseDetails caseData={caseInfo} />
```
Comment (jinkang-0) [inline src/app/cases/page.tsx:228]: not here, but in the `useEffect`, can we also set selectedCard to the UUID of the first card? to make it obvious that the first card is selected by default

File: src/components/CaseDetails/index.tsx [hunk 3]
```diff
@@ -84,6 +88,52 @@ const caseFields = [
...
+  const profile = useContext(ProfileContext);
+
+  const Interest = useMemo(() => {
+    if (profile && profile.userId)
+      return profile.profileReady ? (
+        <InterestForm caseData={caseData} />
+      ) : (
+        <>
+          <H3>
+            Please finish submitting your profile before submitting interest.
```
Comment (jinkang-0) [inline src/components/CaseDetails/index.tsx]: sorry if i misled you LOL
`profileReady` only tells us if the user's profile has been loaded, but not whether or not they have been onboarded
however, it can be useful to setup loading states for the future

File: src/app/cases/page.tsx (hunk 2 above)
Comment (jinkang-0) [reply src/app/cases/page.tsx:228]: we can also make a state that tracks whether or not cases have been loaded, then set that to true in the `.then` handler when we load the data (also in `useEffect`). this lets us handle the loading cases state (could show nothing for now, or leave for later)

File: src/components/CaseDetails/index.tsx (hunk 3 above)
Comment (varortz) [reply src/components/CaseDetails/index.tsx]: OHHHHH hahah ok will update

File: src/components/CaseDetails/styles.tsx (hunk 1 above)
Comment (varortz) [reply src/components/CaseDetails/styles.tsx]: I thought I had gotten rid of all the comments lol, nice catch

**Review by jinkang-0 — APPROVED** [review]
looks good!!

**Review by pragyakallanagoudar — APPROVED** [review]
LOOKS AMAZING!! seems like the only issue is that on the first page load to `/cases` after authentication it still says that the user needs to log in...but https://github.com/calblueprint/immigration-justice-project/pull/56 should fix this?
